index option added for change_table migrations

meinac opened the pull request:
In case if we want to add a column into the existing table
with index on it, we have to add column and index in two
seperate lines.
With this feature we don't need to write an extra line to
add index for column. We can just use `index` option.

Old behaviour in action:

```
  change_table(:languages) do |t|
    t.string :country_code
    t.index: :country_code
  end
```

New behaviour in action:

```
  change_table(:languages) do |t|
    t.string :country_code, index: true
  end
```

Exactly same behaviour is already exist for `create_table` migrations.
There was a [pull request](github.com/rails/rails/pull/20616) created by me which was about adding `index` option for `add_column` helper. I understand the concern behind closing it and it make sense but this addition is different from it. So I'm not pushing same idea again.

gmcgibbon reviewed (commented):
Still relevant in Rails 6 (e925cb4d856088a815bf4a0cf27518d01bb4029d). Approach looks good to me. Please rebase and fix the CHANGELOG conflict and we can look at getting this merged!

gmcgibbon commented:
If there's still pushback, you could refactor this PR to raise an error if someone tries to add an index inline within a `change_table` block but considering you can add indices with `t.index` in `change_table` I see no problem with merging this.

Either way, there's something that needs to be added because its really unclear that this is currently unsupported.

rafaelfranca commented:
I'm fine with it in column. Can you rebase?

meinac committed:
Index option added for change_table migrations

In case if we want to add a column into the existing table
with index on it, we have to add column and index in two
seperate lines.
With this feature we don't need to write an extra line to
add index for column. We can just use `index` option.

Old behaviour in action:
```
  change_table(:languages) do |t|
    t.string :country_code
    t.index: :country_code
  end
```

New behaviour in action:
```
  change_table(:languages) do |t|
    t.string :country_code, index: true
  end
```

Exactly same behaviour is already exist for `create_table` migrations.

meinac commented:
@rafaelfranca done.
